dhis-2/pom.xml

```diff
@@ -854,6 +815,24 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <version>${maven-dependency-plugin.version}</version>
+                <executions>
+                    <execution>
+                        <id>analyze</id>
+                        <goals>
+                            <goal>analyze-only</goal>
+                        </goals>
+                        <configuration>
+                            <failOnWarning>true</failOnWarning>
+                            <skip>${skipTests}</skip>
+                        </configuration>
+                        <phase>test-compile</phase>
```

Comment:
I think it has to come after phase test-compile https://maven.apache.org/plugins/maven-dependency-plugin/analyze-only-mojo.html
that would be process-test-classes https://maven.apache.org/ref/3.8.5/maven-core/lifecycles.html